### PR TITLE
fixing jsdoc

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

**Review by psoleckimoj — APPROVED** [review]
LGTM!

File: packages/eslint-config-hmpps/index.cjs [hunk 1]
```diff
@@ -46,14 +46,18 @@ const scriptExtensionsGlob = '@(js|mjs|cjs|ts|mts|cts)'
 /**
  * Generates the HMPPS shared best-practice eslint rules for typescript projects
  *
- * @param {string[]} extraIgnorePaths add extra glob entires to ignored paths (e.g. build artefacts)
- * @param {string[]} extraPathsAllowingDevDependencies add extra glob entries that should allow dev dependencies (e.g. bin scripts)
- * @param {Linter.Globals} extraGlobals add languageOptions.globals entries (node is already included)
- * @param {Linter.Globals} extraFrontendGlobals add languageOptions.globals entries to frontend assets (browser is already included)
- * @param {Linter.Globals} extraUnitTestGlobals add languageOptions.globals entries to backend server (jest is already included)
- * @param {Linter.Globals} extraIntegrationGlobals add languageOptions.globals entries to frontend assets (cypress, browser and mocha already included)
+ * @typedef { Record<string, Boolean> } Globals
```
Comment (ushkarev) [inline packages/eslint-config-hmpps/index.cjs:49]: clearly i dunno how to write jsdocs properly. intellij seemed content with me referencing the inner argument properties directly 🤷 

also, globals aren't just booleans, but rather `boolean | "off" | "readable" | "readonly" | "writable" | "writeable"` according to `@types/eslint` (which it turns out is now gone!)

Comment (andrewrlee) [reply packages/eslint-config-hmpps/index.cjs:49]: I used wrong type of boolean too

Comment (ushkarev) [reply packages/eslint-config-hmpps/index.cjs:49]: perhaps this? also, intellij now cannot tell the type of anything :( even though this totally appears to be the way the jsdocs [say](https://jsdoc.app/tags-param) you should document object arguments
```
@typedef { import('eslint').Linter.Config } LinterConfig
@typedef { import('eslint').Linter.Globals } LinterGlobals

@param {Object=} args
@param {string[]=} args.extraIgnorePaths add extra glob entires to ignored paths (e.g. build artefacts)
@param {string[]=} args.extraPathsAllowingDevDependencies add extra glob entries that should allow dev dependencies (e.g. bin scripts)
@param {LinterGlobals=} args.extraGlobals add languageOptions.globals entries (node is already included)
@param {LinterGlobals=} args.extraFrontendGlobals add languageOptions.globals entries to frontend assets (browser is already included)
@param {LinterGlobals=} args.extraUnitTestGlobals add languageOptions.globals entries to backend server (jest is already included)
@param {LinterGlobals=} args.extraIntegrationGlobals add languageOptions.globals entries to frontend assets (cypress, browser and mocha already included)

@return {LinterConfig[]}
```

Comment (andrewrlee) [reply packages/eslint-config-hmpps/index.cjs:49]: Should be this: 
` * @typedef { import('eslint').Linter.Globals } Globals`

Comment (ushkarev) [reply packages/eslint-config-hmpps/index.cjs:49]: aha!! the `@typedef` needs to be in a separate comment block and then intellij happily understands it

**Pull request merged**